chatbot/middleware/rate_limiting.py

```
@@ -0,0 +1,368 @@
+"""
+محدودسازی نرخ درخواست (Rate Limiting) برای چت‌بات
+Rate Limiting Middleware for Chatbot
+"""
+
+import time
+from typing import Dict, Optional
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)